File: backend/src/main/java/com/mhbilliards/billiards_management/entity/TableBilliard.java
package com.mhbilliards.billiards_management.entity;

import com.mhbilliards.billiards_management.enums.TableStatus;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Enumerated;
import jakarta.persistence.EnumType;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.FieldDefaults;

@Entity
@Table(name = "table_billiards")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class TableBilliard extends BaseEntity {

    @Column(unique = true, nullable = false)
    String name;

    String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "type_id", nullable = false)
    TableBilliardType type;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    TableStatus status = TableStatus.AVAILABLE;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    Branch branch;

}

File: frontend/src/types/tableBilliard.ts
import { IBaseResponse } from "./base";
import { IBranchResponse } from "./branch";
import { ITableBilliardTypeResponse } from "./tableBilliardType";

export interface ITableBilliardRequest {
    name: string;
    description: string;
    typeId: number | null;
    zoneId: number | null;
    pricePerHour?: number;
    branchId: number | null;
}

export interface ITableBilliardResponse extends IBaseResponse {
    name: string;
    description: string;
    type: ITableBilliardTypeResponse | null;
    pricePerHour: number;
    branch: IBranchResponse | null;
}
